encryption.go

```diff
@@ -13,6 +13,7 @@ import (
 // that produces an encryption key via X25519.
 type X25519Producer interface {
 	X25519EncryptionKey() ([]byte, error)
+	PreviousKey() ([]byte, error)
```

Comment:
I mentioned this on Slack, putting here so it's not forgotten -- the key ID used is usually tied to the current set of credentials and since this is used as AAD the previous key likely needs a different key ID than the current one. So the interface probably needs to be something like `(string, []byte, error)` and after this PR goes in I may try updating X25519EncryptionKey to match.